crates/api_models/src/payments.rs

```diff
@@ -295,38 +368,39 @@ pub struct PaymentsCreateIntentResponse {
         "amount" : 69000
         "product_img_link" : "https://dummy-img-link.com"
     }]"#)]
-    pub order_details: Option<Vec<OrderDetailsWithAmount>>,
+    pub order_details: Option<Vec<pii::SecretSerdeValue>>,
```

Comment:
We should nor move to untyped values for `order_detailsorder_details`, `allowed_payment_method_types`, `connector_metadata`,  `feature_metadata`,